### PR TITLE
Various improvements to support generating recipes for Noetic

### PR DESCRIPTION
**Pull request by shr-project**

### CODE REVIEW

**Comment by shr-project** [comment]
@allenh1 @herb-kuta-lge Please review and merge

File: superflore/generators/bitbake/yocto_recipe.py [hunk 1]
```diff
@@ -91,7 +91,15 @@ def __init__(
             self.license = pkg_fields.upstream_license
             self.description = pkg_fields.description
             self.homepage = pkg_fields.homepage
-            self.build_type = pkg_fields.build_type
+            pkg_build_type = pkg_fields.build_type
+            if pkg_build_type == 'catkin' and \
+               yoctoRecipe._get_ros_version(rosdistro.name) == 2:
+                err("package pkg_name either doesn't have <export><build_type>"
```
Comment (herb-kuta-lge) [inline superflore/generators/bitbake/yocto_recipe.py:97]: `"package pkg_name ..."` -> `"Package {0} ... ".format(pkg_name)`

File: superflore/generators/bitbake/yocto_recipe.py [hunk 2]
```diff
@@ -97,7 +97,8 @@ def __init__(
             self.license = None
             self.homepage = None
             self.build_type = 'catkin' if \
-                yoctoRecipe._get_ros_version(distro) == 1 else 'ament_cmake'
+                yoctoRecipe._get_ros_version(distro.name) == 1 \
```
Comment (herb-kuta-lge) [inline superflore/generators/bitbake/yocto_recipe.py]: Commit body: `get_ros_version` -> `_get_ros_version()`

File: setup.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def append_local_version_label(public_version):
     'docker',
     'pyyaml',
     'pygithub',
-    'catkin_pkg >= 0.4.0',
+    'catkin_pkg >= 0.4.10',
```
Comment (herb-kuta-lge) [inline setup.py:64]: Commit summary: `get_build_type` -> `get_build_type()`

Commit body: `0.4.10 verion` -> `v0.4.10`

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]

File: superflore/generators/bitbake/yocto_recipe.py [hunk 4]
```diff
@@ -562,7 +562,7 @@ def generate_ros_distro_inc(
                 conf_file.write('ROS_VERSION = "{}"\n'.format(ros_version))
                 conf_file.write('# DO NOT OVERRIDE ROS_PYTHON_VERSION\n')
                 ros_python_version = 3
-                if ros_version == 1:
+                if distro == 'melodic':
```
[the remark this answers is not in the document]
Comment (shr-project) [reply superflore/generators/bitbake/yocto_recipe.py]: ros-generate-{cache,recipes}.sh does abort.

**Comment by shr-project** [comment]
Depends on https://github.com/ros-infrastructure/superflore/pull/281 now

**Review by allenh1 — DISMISSED** [review]
lgtm

**Comment by allenh1** [comment]
@herb-kuta-lge please take a look at this one when you can

File: superflore/generators/bitbake/yocto_recipe.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (herb-kuta-lge) [reply superflore/generators/bitbake/yocto_recipe.py]: OK, then at least document that ROS 1 distros prior to *melodic* are not supported in `README.md`:
```
  --ros-distro ROS_DISTRO
                        regenerate packages for the specified distro
```
Append `; ROS 1 distros prior to "melodic" are not supported` (in two places).

File: superflore/generators/bitbake/yocto_recipe.py (hunk 2 above)
Comment (herb-kuta-lge) [reply superflore/generators/bitbake/yocto_recipe.py]: There's another one to change in the commit body of 100feaea0e34ff469fc429345a7f061fcd49b458 .

File: setup.py (hunk 3 above)
Comment (herb-kuta-lge) [reply setup.py]: Commit summary: get_build_type -> get_build_type() in a409be822b8e983dff5af518bb5a61c21a18ce0a (just a nit, obviously).

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]
I've unresolved 3 of the conversations.

File: superflore/generators/bitbake/yocto_recipe.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (shr-project) [reply superflore/generators/bitbake/yocto_recipe.py]: Which 2 places? This section of README.md is matching with the -h output.

**Comment by shr-project** [comment]
Please DON'T SQUASH the commits when merging, it will be impossible to review and understand the commits after they are squashed.

File: superflore/generators/bitbake/yocto_recipe.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (herb-kuta-lge) [reply superflore/generators/bitbake/yocto_recipe.py]: > Which 2 places?

There aren't (which is why I edited my comment :-) ). But I did forget that this section is the output from `-h`. So instead, add this note to the **Common Usage:** section after `The --ros-distro flag must be supplied.`: `ROS 1 distros prior to "melodic" are not supported`.

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]
Unresolved 1 conversation.

**Review by herb-kuta-lge — DISMISSED** [review]
LGTM.

**Review by allenh1 — APPROVED** [review]

**Review by herb-kuta-lge — APPROVED** [review]

**Pull request merged**